src/backend/distributed/utils/resource_lock.c

```diff
@@ -272,6 +272,28 @@ LockShardListMetadata(List *shardIntervalList, LOCKMODE lockMode)
 	}
 }
 
+/* LockPlacementListMetadata takes locks on the metadata of all shards in
+ * shardPlacementList to prevet concurrent placement changes.
+ */
+void
+LockShardPlacementListMetadata(List *shardPlacementList, LOCKMODE lockMode)
```

Comment:
I think this function is a bit confusing because it suggests that we lock shard placements, but we don't lock shard placements, we lock shard IDs. 

What if we introduce something like a `ShardIntervalsForPlacementList` to convert a placement list to a shard interval list, such that we can use the original API?

Reply:
I've improved the naming of the function to LockShardsInPlacementListMetadata, in the hope that it would clear confusion enough to be acceptable. I though about converting list formats and keep using the existing API, but I'm not convinced that's the better approach: after all we are indeed processing a list of placements which have ShardIds in there.